docs/en/sql-reference/functions/json-functions.md

```diff
@@ -843,6 +843,198 @@ JSONExtractRaw(json [, indices_or_keys]...)
+SELECT JSONExtractStringCaseInsensitive('{"User": {"Name": "John"}}', 'user', 'name') = 'John'
+```
+
+### JSONExtractCaseInsensitive {#jsonextractcaseinsensitive}
+
+Parses JSON and extracts a value of the given ClickHouse data type using case-insensitive key matching.
+
+**Syntax**
+
+```sql
+JSONExtractCaseInsensitive(json [, indices_or_keys...], return_type)
+```
+
+**Example**
+
+```sql
+SELECT JSONExtractCaseInsensitive('{"Number": 123}', 'number', 'Int32') = 123
+SELECT JSONExtractCaseInsensitive('{"List": [1, 2, 3]}', 'list', 'Array(Int32)') = [1, 2, 3]
+```
+
+### JSONExtractKeysAndValuesCaseInsensitive {#jsonextractkeysandvaluescaseinsensitive}
+
+Parses key-value pairs from JSON using case-insensitive key matching.
+
+**Syntax**
+
+```sql
+JSONExtractKeysAndValuesCaseInsensitive(json [, indices_or_keys...], value_type)
+```
+
+**Example**
+
+```sql
+SELECT JSONExtractKeysAndValuesCaseInsensitive('{"Name": "Alice", "AGE": 30}', 'String')[1] = ('Name', 'Alice')
+```
+
+### JSONExtractRawCaseInsensitive {#jsonextractrawcaseinsensitive}
+
+Returns part of the JSON as an unparsed string using case-insensitive key matching.
+
+**Syntax**
+
+```sql
+JSONExtractRawCaseInsensitive(json [, indices_or_keys]...)
+```
+
+**Example**
+
+```sql
+SELECT JSONExtractRawCaseInsensitive('{"Object": {"key": "value"}}', 'OBJECT') = '{"key":"value"}'
+```
+
+### JSONExtractArrayRawCaseInsensitive {#jsonextractarrayrawcaseinsensitive}
+
+Returns an array with elements of JSON array, each represented as unparsed string, using case-insensitive key matching.
+
+**Syntax**
+
+```sql
+JSONExtractArrayRawCaseInsensitive(json [, indices_or_keys]...)
+```
+
+**Example**
+
+```sql
+SELECT JSONExtractArrayRawCaseInsensitive('{"Items": [1, 2, 3]}', 'ITEMS') = ['1', '2', '3']
+```
+
+### JSONExtractKeysAndValuesRawCaseInsensitive {#jsonextractkeysandvaluesrawcaseinsensitive}
+
+Extracts raw key-value pairs from JSON using case-insensitive key matching.
+
+**Syntax**
+
+```sql
+JSONExtractKeysAndValuesRawCaseInsensitive(json [, indices_or_keys]...)
+```
+
+**Example**
+
+```sql
+SELECT JSONExtractKeysAndValuesRawCaseInsensitive('{"Name": "Alice", "AGE": 30}')[1] = ('Name', '"Alice"')
+```
+
+### JSONExtractKeysCaseInsensitive {#jsonextractkeyescaseinsensitive}
+
+Parses a JSON string and extracts the keys using case-insensitive key matching to navigate to nested objects.
+
+**Syntax**
+
+```sql
+JSONExtractKeysCaseInsensitive(json [, indices_or_keys]...)
+```
+
+**Example**
+
+```sql
+SELECT JSONExtractKeysCaseInsensitive('{"Name": "Alice", "AGE": 30}') = ['Name', 'AGE']
+SELECT JSONExtractKeysCaseInsensitive('{"User": {"name": "John", "AGE": 25}}', 'user') = ['name', 'AGE']
+```
+
+**Implementation Notes**
+
+- When multiple keys match with different cases, the first match is returned
+- Case-insensitive matching only applies to object keys, not to array indices or the extracted values
+- The comparison is ASCII case-insensitive
```

Comment:
This is kind of important information in my opinion, so let's mention this in the first block.

Reply:
I took this whole section out and included the relevant info in the first block.